core-plugins/src/main/java/co/cask/hydrator/plugin/batch/source/FileSourceConfig.java

```diff
@@ -155,15 +149,28 @@ protected void validate() {
       (format.equalsIgnoreCase("avro")) || format.equalsIgnoreCase("parquet"))) {
       throw new IllegalArgumentException("Format can only be 'text', 'avro' or 'parquet'");
     }
-    if (!format.equalsIgnoreCase("text")) {
-      try {
-        new Schema.Parser().parse(inputSchema);
-      } catch (Exception e){
-        throw new IllegalArgumentException("Unable to parse schema with error: " + e.getMessage(), e);
+    Schema parsedSchema;
+    try {
+      parsedSchema = Schema.parseJson(schema);
+    } catch (Exception e) {
+      throw new IllegalArgumentException("Unable to parse schema with error: " + e.getMessage(), e);
+    }
+
+    if (format.equalsIgnoreCase("text")) {
+      if (schema != null && !schema.isEmpty()) {
```

Comment:
combine this with the previous line to have  `"text".equalsIgnoreCase(format) && parsedSchema != null`